log4j-core/src/main/java/org/apache/logging/log4j/core/config/Property.java

```diff
@@ -57,6 +63,14 @@ public String getName() {
         return name;
     }
 
+    /**
+     * Returns the original raw property value without substitution.
+     * @return the raw value of the property, or null if it is not set.
+     */
+    public String getRawValue() {
```

Comment:
I'm not thrilled about adding these new methods because they muddy the waters between the object model and configuration node->plugin api.

Ideally we could add a parameter to the `@PluginElement("Properties") final Property[] properties` parameter in `PropertiesPlugin` to opt out of all substitutions when that data is read, allowing us to produce the same behavior. When I looked into that, it didn't seem like something the plugin system is built to handle (contextual configuration).

Reply:
Ok